src/main/java/org/jabref/gui/util/DefaultTaskExecutor.java

```diff
@@ -96,7 +97,11 @@ public static void runInJavaFXThread(Runnable runnable) {
 
     @Override
     public <V> Future<V> execute(BackgroundTask<V> task) {
-        return execute(getJavaFXTask(task));
+        Task<V> javafxTask = getJavaFXTask(task);
+        if (task.showToUser()) {
+            Globals.stateManager.addBackgroundTask(javafxTask);
```

Comment:
We try to remove the `Globals` class. Is it possible to insert the `stateManager` via a constructor parameter?

Reply:
Yes, that should be easy.

Reply:
Fixed in https://github.com/JabRef/jabref/pull/6443/commits/2c9ccea66fef2bcc6e68d0fadfdf407b1a0b6262